balancer/weightedtarget/weightedtarget_test.go

```diff
@@ -1218,3 +1218,77 @@ func (s) TestInitialIdle(t *testing.T) {
 		t.Fatalf("Received aggregated state: %v, want Idle", state)
 	}
 }
+
+type updateStateBalancerBuilder struct{}
+
+func (updateStateBalancerBuilder) Build(cc balancer.ClientConn, opts balancer.BuildOptions) balancer.Balancer {
+	return &updateStateBalancer{cc: cc}
+}
+
+const updateStateBalancerName = "update_state_balancer"
+
+func (updateStateBalancerBuilder) Name() string {
+	return updateStateBalancerName
+}
+
+func init() {
+	balancer.Register(updateStateBalancerBuilder{})
+}
+
+type updateStateBalancer struct {
+	cc balancer.ClientConn
+	balancer.Balancer
+}
+
+func (b *updateStateBalancer) UpdateClientConnState(s balancer.ClientConnState) error {
+	b.cc.UpdateState(balancer.State{ConnectivityState: connectivity.TransientFailure, Picker: nil})
+	b.cc.UpdateState(balancer.State{ConnectivityState: connectivity.Ready, Picker: nil})
+	return nil
+}
+
+func (b *updateStateBalancer) Close() {}
+
```

Comment:
Can we replace this with a stub balancer?

Reply:
Oh yes, good call.  Done.